test/e2e/helpers.go

```diff
@@ -911,3 +913,77 @@ func WaitForDaemonset(ctx context.Context, input clusterctl.ApplyClusterTemplate
 		return false
 	}, input.WaitForControlPlaneIntervals...).Should(Equal(true))
 }
+
+// podListHasNumPods fulfills the Cluster API PodListCondition type spec
+// given a list of pods, we validate for an exact number of those pods in a Running state
+func podListHasNumPods(numPods int) func(pl *corev1.PodList) error {
+	return func(pl *corev1.PodList) error {
+		var runningPods int
+		for _, p := range pl.Items {
+			if p.Status.Phase == corev1.PodRunning {
+				runningPods++
+			}
+		}
+		if runningPods != numPods {
+			return errors.Errorf("expected %d Running pods, got %d", numPods, runningPods)
+		}
+		return nil
+	}
+}
+
+// podListHasAtLeastNumPods fulfills the cluster-api PodListCondition type spec
+// given a list of pods, we validate for at least 1 number of those pods in a Running state
+func podListHasAtLeastNumPods(numPods int) func(pl *corev1.PodList) error {
+	return func(pl *corev1.PodList) error {
+		var runningPods int
+		for _, p := range pl.Items {
+			if p.Status.Phase == corev1.PodRunning {
+				runningPods++
+			}
+		}
+		if runningPods >= numPods {
+			return errors.Errorf("expected %d Running pods, got %d", numPods, runningPods)
+		}
+		return nil
+	}
+}
+
+func defaultConfigCluster(clusterName, namespace string) clusterctl.ConfigClusterInput {
```

Comment:
Thoughts about reusing this in more places? It's a good idea to refactor lines of code that come up often but if we're not using it everywhere it can lead to inconsistencies

Reply:
At the moment, it is not used elsewhere. But thought, would be a good start to start using this. I figured out that this make the e2e code more readable and make it really to understand what the test is doing on a high level. 
I can file up an issue so that we start using this defaultConfigCluster at other places in e2e and I can drive that to completion. 
My ideas is the following: 
- Get the defaultConfig using this function in our e2e test
- If we want to add/override -- do this e.g. `configCluster.WorkerMachineCount = pointer.Int64Ptr(3)`
wdyt?